### PR TITLE
vDPA: Add utilities for vDPA test

### PR DESCRIPTION
**Pull request by Yingshun**
Add support to setup vDPA test environment.
BTW, there is no 'net' dir for the vDPA env, so update to read the
link of the VF in get_vf_pci_id() to get the pci id.

Signed-off-by: Yingshun Cui <yicui@redhat.com>
Test results: https://github.com/autotest/tp-libvirt/pull/3838#issue-1035947106

### CODE REVIEW

File: virttest/utils_vdpa.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        """
+        Add vDPA device
+
+        :param idx: Index of dev
+        :param pci_addr: PCI address
+        """
+        cmd = "vdpa dev add name vdpa{} mgmtdev pci/{}".format(idx, pci_addr)
+        process.run(cmd, shell=True)
+
+    def set_dev_manged_no(self, dev):
```
Comment (kamilvarga) [inline virttest/utils_vdpa.py:97]: Would be better to cal function set_dev_managed_no, what do you think?

**Review by kamilvarga — COMMENTED** [review]
Others LGTM.

File: virttest/utils_vdpa.py (hunk 1 above)
Comment (Yingshun) [reply virttest/utils_vdpa.py]: Oops! Updated. Thank you!

**Review by kamilvarga — APPROVED** [review]
LGTM.

File: virttest/utils_switchdev.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    for idx in range(vf_no):
+        pci_addr = utils_sriov.get_vf_pci_id(pf_pci, idx)
+        cmd = "echo %s >  /sys/bus/pci/drivers/mlx5_core/bind" % pci_addr
+        process.run(cmd, shell=True)
+
+
+def get_switchid(interface):
+    """
+    Get switch id
+
```
Comment (dzhengfy) [inline virttest/utils_switchdev.py:57]: Docstring for :return:

File: virttest/utils_switchdev.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    :param interface: interface name
+    """
+    cmd = "ip -d link show %s | sed -n 's/.* switchid \([^ ]*\).*/\\1/p'" % interface
+    return process.run(cmd, shell=True).stdout_text.strip()
+
+
+def get_all_representors():
+    """
+    Get all representors from '/sys/class/net' dir
```
Comment (dzhengfy) [inline virttest/utils_switchdev.py:66]: Docstring for :return:

File: virttest/utils_switchdev.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        except OSError:
+            pass
+        else:
+            reps[ifc].update({'switch_id': switch_id})
+    return reps
+
+
+def get_representor(reps, vf_idx, switch_id):
+    """
+    Get representor by given vf_idx and switch_id
```
Comment (dzhengfy) [inline virttest/utils_switchdev.py:89]: Docstring for :return: and :param xx:

File: virttest/utils_switchdev.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    """
+    for ifc, info in reps.items():
+        if info.get('switch_id') == switch_id and \
+           info.get('port') == "pf0vf" + str(vf_idx):
+            return ifc
+
+
+def get_rep_list(pf_iface, vf_no=4):
+    """
+    Get representor list by given pf iface and vf number
```
Comment (dzhengfy) [inline virttest/utils_switchdev.py:99]: docstring for return and param

File: virttest/utils_vdpa.py [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+        Unload modules
+        """
+        modules = ['mlx5_vdpa', 'vhost_vdpa', 'vdpa']
+        for module_name in modules:
+            KernelModuleHandler(module_name).unload_module()
+
+    def config_vdpa_dev(self):
+        """
+        Bind VFs and add vDPA dev
+        """
```
Comment (dzhengfy) [inline virttest/utils_vdpa.py:61]: docstring for :raises:

File: virttest/utils_vdpa.py [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+            if not vf_dev:
+                raise exceptions.TestError("Cannot get VF network device!")
+            self.set_tc_offload(vf_dev)
+            self.set_dev_managed_no(vf_dev)
+            self.add_vdpa_dev(idx, vf_pci)
+
+    def get_rep_list(self):
+        """
+        Get representor list
+        """
```
Comment (dzhengfy) [inline virttest/utils_vdpa.py:75]: docstring for :return:

File: virttest/utils_vdpa.py [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+        Cleanup vDPA Simulator environment
+        """
+        self.unload_modules()
+        LOG.info("vDPA Simulator environment recover successfully.")
+
+
+def get_vdpa_pci(driver='mlx5_core'):
+    """
+    Get PF's pci id by given driver
```
Comment (dzhengfy) [inline virttest/utils_vdpa.py:244]: docstring for return and param

File: virttest/utils_switchdev.py (hunk 2 above)
Comment (Yingshun) [reply virttest/utils_switchdev.py:57]: done

File: virttest/utils_switchdev.py (hunk 3 above)
Comment (Yingshun) [reply virttest/utils_switchdev.py:67]: added

File: virttest/utils_switchdev.py (hunk 4 above)
Comment (Yingshun) [reply virttest/utils_switchdev.py:92]: updated

File: virttest/utils_switchdev.py (hunk 5 above)
Comment (Yingshun) [reply virttest/utils_switchdev.py:106]: updated.

File: virttest/utils_vdpa.py (hunk 6 above)
Comment (Yingshun) [reply virttest/utils_vdpa.py:63]: done.

File: virttest/utils_vdpa.py (hunk 8 above)
Comment (Yingshun) [reply virttest/utils_vdpa.py:248]: Done.

File: virttest/utils_vdpa.py (hunk 7 above)
Comment (Yingshun) [reply virttest/utils_vdpa.py:79]: Updated.

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**